libshaderc_spvc/src/spvcir_pass.cc

```diff
@@ -23,25 +23,46 @@ SpvcIrPass::SpvcIrPass(spirv_cross::ParsedIR *ir) {
   ir_ = ir;
   current_function_ = nullptr;
   current_block_ = nullptr;
+  status_ = Status::SuccessWithoutChange;
 
   auto s = ir_->spirv.data();
-  assert(ir_->spirv.size() > 3 && "spirv data is too small");
+  CheckAndSetErrorMessage(ir_->spirv.size() > 3,
+                          "SpvcIrPass: spirv data is too small");
   uint32_t bound = s[3];
   ir_->set_id_bounds(bound);
 }
 
+void SpvcIrPass::CheckAndSetErrorMessage(bool condition, std::string message) {
```

Comment:
Maybe CheckConditionAndSetErrorMessage

Reply:
Also maybe also return a bool that is condition, so that you can write thing like 

`if (!CheckConditonAndSetErrorMessage(foo, "I prefer bar over foo") return;`

Reply:
Also probably work putting a comment on this method to say what it is trying to do at a high-level.

`// Check boolean condition. If false record the error message, and set pass status to failure to stop processing.`